### PR TITLE
doc: Add an example for target units

### LINKED ISSUE #67: add example for creating your own target unit that builds on top of a well-known target like rescue.target or so
I have a bunch of headless servers that sometimes need manual fsck of the /
It would be really nice to have an "emergency with networking" option, if possible

Without it, it's required to connect a KVM every time this happens...

### PR DESCRIPTION
**Pull request by dasJ**
This adds an example target unit to the man page of systemd targets.

Closes #67.

### CODE REVIEW

File: man/systemd.target.xml [hunk 1]
```diff
@@ -99,6 +99,24 @@
...
+      <title>Simple target</title>
+
+      <programlisting># emergency-net.target
+[Unit]
+Description=Emergency with networking
+Requires=emergency.target
+After=emergency.target
+AllowIsolate=yes</programlisting>
+
+      <para>Units can now have a <varname>WantedBy=</varname> value
```
Comment (keszybz) [inline man/systemd.target.xml]: That part is misleading. WantedBy can only appear in the [Install] section, so it'd be better to reword this as "Other units can have WantedBy=emergency-net.target in the [Install] section. After they are enabled, they will be started before emergency-net.target is started."

Also, shouldn't your emergency-net.target depend on some networking units? Right now the example seems incomplete, because it doesn't yet differ from emergency.target in any way.

Also, since this target seems to be designed to be used as emergency.target replacement, it'd be nice to give an example of how to actually use it.

**Comment by arvidjaar** [comment]
> Also, shouldn't your emergency-net.target depend on some networking units?

It is impossible without effectively re-implementing or at least making significant changes to systemd. The whole point of `emergency.target` is to be replacement for normal startup sequence, while network units do depend on normal startup sequence.

**Comment by keszybz** [comment]
What about `systemd-networkd.service`? It has DefaultDependencies=no...

**Comment by dasJ** [comment]
Updated the PR with more information about `DefaultDependencies`, how to start the unit, and a dependency to`systemd-networkd.service`.

File: man/systemd.target.xml [hunk 2]
```diff
@@ -99,6 +99,36 @@
...
+Description=Emergency with networking
+Requires=emergency.target systemd-networkd.target
+After=emergency.target systemd-networkd.target
+AllowIsolate=yes</programlisting>
+
+      <para>When adding dependencies to other units, it's important to check if they set
+      <varname>DefaultDependencies=no</varname>. If they don't, they depend on the
+      <varname>WantedBy=</varname> value from their <varname>[Install]</varname>
+      section and therefore pull the target (and possibly the entire boot sequence)
+      as dependency.</para>
```
Comment (keszybz) [inline man/systemd.target.xml]: This paragraph is wrong. [Install] is only used during installation. Maybe just remove the whole para.

**Comment by keszybz** [comment]
I updated the text a bit.
